src/header.rs

```diff
@@ -68,32 +73,70 @@ pub fn parse_header(rlp: &mut CoreParser) -> Result<Header, ParseError> {
     })
 }
 
-fn parse_rdict(rlp: &mut CoreParser) -> Result<Vec<(String, HeaderProp)>, ParseError> {
-    // Other the actual network data, the header property associative array is the hardest to parse.
-    // The format is to:
-    // - Read string
-    // - If string is "None", we're done
-    // - else we're dealing with a property, and the string just read is the key. Now deserialize the
-    //   value.
-    // The return type of this function is a key value vector because since there is no format
-    // specification, we can't rule out duplicate keys. Possibly consider a multi-map in the future.
+#[derive(Clone, Copy)]
+enum ParserMode {
+    Standard,
+    Quirks,
+}
 
+fn parse_rdict(
+    rlp: &mut CoreParser,
+    mode: ParserMode,
+) -> Result<Vec<(String, HeaderProp)>, ParseError> {
+    // The return type of this function is a key value vector because since there is no format
+    // specification, we can't rule out duplicate keys.
     let mut res: Vec<_> = Vec::new();
     loop {
         let key = rlp.parse_str()?;
-        if key == "None" || key == "\0\0\0None" {
+        if key == "None" {
             break;
         }
 
-        let val = match rlp.parse_str()? {
-            "ArrayProperty" => decode_prop(rlp, array_property),
-            "BoolProperty" => decode_prop(rlp, bool_property),
-            "ByteProperty" => decode_prop(rlp, byte_property),
-            "FloatProperty" => decode_prop(rlp, float_property),
-            "IntProperty" => decode_prop(rlp, int_property),
-            "NameProperty" => decode_prop(rlp, name_property),
-            "QWordProperty" => decode_prop(rlp, qword_property),
-            "StrProperty" => decode_prop(rlp, str_property),
+        let kind = rlp.parse_str()?;
+        let size = rlp.take(8, le_u64)? as usize;
+        let val = match kind {
+            "BoolProperty" => match mode {
+                // The size SHOULD be zero, but we're ignoring it.
+                ParserMode::Standard => rlp.take_data(1).map(bool_prop),
+                ParserMode::Quirks => rlp.take_data(4).map(bool_prop),
+            },
+            "ByteProperty" => match mode {
+                ParserMode::Standard => {
+                    let kind = rlp.parse_str()?;
```

Comment:
For what it's worth, the `kind` is always `"OnlinePlatform"`. For standard decoding it certainly seems like they intended for this to be an arbitrary key-value pair. But for quirks decoding I think it makes more sense to set the `kind` to `"OnlinePlatform"` and then set the value to whatever gets decoded (usually something like `"OnlinePlatform_Steam"`).

Reply:
Yeah that's true, let me add in a debug assert and make a note of that for when I want to make a breaking change to the output